diff --git a/configure_kubernetes.sh b/configure_kubernetes.sh
new file mode 100755
index 0000000..5abd1da
--- /dev/null
+++ ./configure_kubernetes.sh
@@ -0,0 +1,14 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+kubeadm init --pod-network-cidr=10.244.0.0/16
+swapoff -a
+mkdir -p $HOME/.kube
+cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+chown $(id -u):$(id -g) $HOME/.kube/config
+
+kubectl taint nodes --all node-role.kubernetes.io/master-
+kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
diff --git a/install.sh b/install.sh
new file mode 100755
index 0000000..68cbafa
--- /dev/null
+++ ./install.sh
@@ -0,0 +1,12 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+apt install -y curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add
+snap install kube-controller-manager
+
+apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
+apt install -y kubeadm
+swapoff -a
diff --git a/installation.sh b/installation.sh
new file mode 100755
index 0000000..cbc3006
--- /dev/null
+++ ./installation.sh
@@ -0,0 +1,9 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.9.6/manifests/namespace.yaml
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.9.6/manifests/metallb.yaml
+# On first install only
+kubectl create secret generic -n metallb-system memberlist --from-literal=secretkey="$(openssl rand -base64 128)"
diff --git a/metallb-config.yaml b/metallb-config.yaml
new file mode 100644
index 0000000..07c4b65
--- /dev/null
+++ ./metallb-config.yaml
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    address-pools:
+    - name: default
+      protocol: layer2
+      addresses:
+      - 192.168.31.240-192.168.31.250
diff --git a/metallb_all.sh b/metallb_all.sh
new file mode 100755
index 0000000..8a89921
--- /dev/null
+++ ./metallb_all.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./preparation.sh
+./installation.sh
+./metallb_configure.sh
diff --git a/metallb_configure.sh b/metallb_configure.sh
new file mode 100755
index 0000000..d8f70b5
--- /dev/null
+++ ./metallb_configure.sh
@@ -0,0 +1,6 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+kubectl apply -f ./metallb-config.yaml
diff --git a/openebs_all.sh b/openebs_all.sh
new file mode 100755
index 0000000..1a5e274
--- /dev/null
+++ ./openebs_all.sh
@@ -0,0 +1,9 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./openebs_prerequisites.sh
+./openebs_set_cluster_admin_user_context.sh
+kubectl apply -f https://openebs.github.io/charts/openebs-operator.yaml
+./openebs_verify.sh
diff --git a/openebs_prerequisites.sh b/openebs_prerequisites.sh
new file mode 100755
index 0000000..189ae58
--- /dev/null
+++ ./openebs_prerequisites.sh
@@ -0,0 +1,10 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+apt-get update
+apt-get install -y open-iscsi
+systemctl enable --now iscsid
+cat /etc/iscsi/initiatorname.iscsi
+systemctl status iscsid
diff --git a/openebs_set_cluster_admin_user_context.sh b/openebs_set_cluster_admin_user_context.sh
new file mode 100755
index 0000000..a397b59
--- /dev/null
+++ ./openebs_set_cluster_admin_user_context.sh
@@ -0,0 +1,8 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+# ./.kube/config
+kubectl config set-context admin-ctx --cluster=kubernetes --user=kubernetes-admin
+kubectl config use-context admin-ctx
diff --git a/openebs_verify.sh b/openebs_verify.sh
new file mode 100755
index 0000000..41e02b7
--- /dev/null
+++ ./openebs_verify.sh
@@ -0,0 +1,10 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+kubectl get pods -n openebs
+kubectl get sc
+kubectl get blockdevice -n openebs
+# kubectl describe blockdevice blockdevice-c3f4da0efa6b0330fde5a02647a5f8df -n openebs
+kubectl get sp
diff --git a/preparation.sh b/preparation.sh
new file mode 100755
index 0000000..8cf9675
--- /dev/null
+++ ./preparation.sh
@@ -0,0 +1,14 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+# see what changes would be made, returns nonzero returncode if different
+kubectl get configmap kube-proxy -n kube-system -o yaml | \
+    sed -e "s/strictARP: false/strictARP: true/" | \
+    kubectl diff -f - -n kube-system
+
+# actually apply the changes, returns nonzero returncode on errors only
+kubectl get configmap kube-proxy -n kube-system -o yaml | \
+    sed -e "s/strictARP: false/strictARP: true/" | \
+    kubectl apply -f - -n kube-system
diff --git a/uninstall.sh b/remove_kubernetes_config.sh
similarity index 100%
rename from uninstall.sh
rename to remove_kubernetes_config.sh
diff --git a/reset.sh b/reset.sh
index 02f2f79..6beedac 100755
--- ./reset.sh
+++ ./reset.sh
@@ -3,13 +3,7 @@ SCRIPT=$(realpath "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
-./uninstall.sh
-kubeadm init --pod-network-cidr=10.244.0.0/16
-mkdir -p $HOME/.kube
-cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
-chown $(id -u):$(id -g) $HOME/.kube/config
-kubectl taint nodes --all node-role.kubernetes.io/master-
-kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
-# kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
+./remove_kubernetes_config.sh
+./configure_kubernetes.sh
 
-#./configure_kubernetes_dynamic_volume_provisioning.sh
+./openebs_all.sh
diff --git a/setup.sh b/setup.sh
index 7dca79c..1f94bbf 100755
--- ./setup.sh
+++ ./setup.sh
@@ -3,27 +3,8 @@ SCRIPT=$(realpath "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
-curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add
-apt install curl
-snap install kube-controller-manager
+./install.sh
+./configure_kubernetes.sh
 
-apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
-apt install -y kubeadm && \
-    swapoff -a && \
-    kubeadm init --pod-network-cidr=10.244.0.0/16
-
-swapoff -a
-mkdir -p $HOME/.kube
-cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
-chown $(id -u):$(id -g) $HOME/.kube/config
-
-kubectl taint nodes --all node-role.kubernetes.io/master-
-kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
-#kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
-
-# ./setup_gluster.sh
-# ../examples/volumes/glusterfs/deploy.sh
-# ../examples/staging/persistent-volume-provisioning/README.md
-
-# ./setup_heketi.sh
-# configure_kubernetes_dynamic_volume_provisioning.sh
+./openebs_all.sh
+./metallb_all.sh
