diff --git a/configure_kubernetes.sh b/configure_kubernetes.sh
new file mode 100755
index 0000000..5abd1da
--- /dev/null
+++ ./configure_kubernetes.sh
@@ -0,0 +1,14 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+kubeadm init --pod-network-cidr=10.244.0.0/16
+swapoff -a
+mkdir -p $HOME/.kube
+cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+chown $(id -u):$(id -g) $HOME/.kube/config
+
+kubectl taint nodes --all node-role.kubernetes.io/master-
+kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+# kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
diff --git a/install.sh b/install.sh
new file mode 100755
index 0000000..68cbafa
--- /dev/null
+++ ./install.sh
@@ -0,0 +1,12 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+apt install -y curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add
+snap install kube-controller-manager
+
+apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
+apt install -y kubeadm
+swapoff -a
diff --git a/uninstall.sh b/remove_kubernetes_config.sh
similarity index 100%
rename from uninstall.sh
rename to remove_kubernetes_config.sh
diff --git a/reset.sh b/reset.sh
index 02f2f79..ae02b1f 100755
--- ./reset.sh
+++ ./reset.sh
@@ -3,13 +3,5 @@ SCRIPT=$(realpath "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
-./uninstall.sh
-kubeadm init --pod-network-cidr=10.244.0.0/16
-mkdir -p $HOME/.kube
-cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
-chown $(id -u):$(id -g) $HOME/.kube/config
-kubectl taint nodes --all node-role.kubernetes.io/master-
-kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
-# kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
-
-#./configure_kubernetes_dynamic_volume_provisioning.sh
+./remove_kubernetes_config.sh
+./configure_kubernetes.sh
diff --git a/setup.sh b/setup.sh
index 7dca79c..766c27a 100755
--- ./setup.sh
+++ ./setup.sh
@@ -3,27 +3,5 @@ SCRIPT=$(realpath "$0")
 SCRIPTPATH=$(dirname "$SCRIPT")
 cd "$SCRIPTPATH"
 
-curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add
-apt install curl
-snap install kube-controller-manager
-
-apt-add-repository "deb http://apt.kubernetes.io/ kubernetes-xenial main"
-apt install -y kubeadm && \
-    swapoff -a && \
-    kubeadm init --pod-network-cidr=10.244.0.0/16
-
-swapoff -a
-mkdir -p $HOME/.kube
-cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
-chown $(id -u):$(id -g) $HOME/.kube/config
-
-kubectl taint nodes --all node-role.kubernetes.io/master-
-kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
-#kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.0/aio/deploy/recommended.yaml
-
-# ./setup_gluster.sh
-# ../examples/volumes/glusterfs/deploy.sh
-# ../examples/staging/persistent-volume-provisioning/README.md
-
-# ./setup_heketi.sh
-# configure_kubernetes_dynamic_volume_provisioning.sh
+./install.sh
+./configure_kubernetes.sh
